src/hotspot/share/opto/locknode.cpp

```diff
@@ -189,6 +189,17 @@ void Parse::do_monitor_enter() {
   // the monitor object is not part of debug info expression stack
   pop();
 
+  if (DoPartialEscapeAnalysis) {
+    // TODO: PEA can support monitors
+    PartialEscapeAnalysis* pea = PEA();
+    PEAState& state = jvms()->alloc_state();
+    ObjID id;
+
+    if ((id = pea->is_alias(obj)) && state.get_object_state(id)->is_virtual()) {
```

Comment:
we can't check is_virtual() like this.  I encountered a bug before. 
now **aliases** is a global data structure, or it's flow-insensitive. In contrast, Allocation state is still flow-sensitive. It's possible that 'obj' is an alias but the object doesn't exist in current basic block. 

you can check is_virtual() using this expression:

```java
state.as_virtual(pea, obj) != nullptr
```